### PR TITLE
faas-cli 0.4.5

### PR DESCRIPTION
**Pull request by alexellis**
* Bump version and inject SHA into version ldflags.

Signed-off-by: Alex Ellis <alexellis2@gmail.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by alexellis** [comment]
Hi, this was previously handled by @johnmccabe .. I'm wanting to bump the version and to match the new version of the CLI which injects a version / SHA at build-time.

**Comment by alexellis** [comment]
Ping @ilovezfs

**Comment by ilovezfs** [comment]
@alexellis I think you'll want to build from the tag instead of the tarball if the git commit needs to be in there. The SHA-256 of the source archive tarball is the not the same thing as a commit hash. You can take a look at the `docker` formula and the `gitlab-ci-multi-runner` formula for examples.

**Comment by alexellis** [comment]
Thanks for the comment. By downloading the src tarball we don't have any sort of Git access, correct? At the moment I'm mainly looking at putting a unique ID into the CLI built by brew. The Git Commit ID would be better.

**Comment by ilovezfs** [comment]
These lines:
https://github.com/Homebrew/homebrew-core/blob/master/Formula/docker.rb#L4-L6
https://github.com/Homebrew/homebrew-core/blob/master/Formula/docker.rb#L31-L35

**Comment by alexellis** [comment]
@ilovezfs this now matches the build process via https://github.com/alexellis/faas-cli/blob/master/Dockerfile.redist

File: Formula/faas-cli.rb [hunk 1]
```diff
@@ -1,8 +1,9 @@
 class FaasCli < Formula
   desc "CLI for templating and/or deploying FaaS functions"
   homepage "http://docs.get-faas.com/"
-  url "https://github.com/alexellis/faas-cli/archive/0.4.tar.gz"
-  sha256 "f7ecebde2545243e9f37f7feb9fc2a171585d3f9e7998f981611f038bbc93987"
+  url "https://github.com/alexellis/faas-cli.git",
+    :tag => "0.4.5",
```
Comment (ilovezfs) [inline Formula/faas-cli.rb]: The `:` should be directly under the `"`.

File: Formula/faas-cli.rb [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
+    :revision => "e9c738f9a2833980a1bfc0102b3dfdf0cce2684a"
```
Comment (ilovezfs) [inline Formula/faas-cli.rb]: The `:` should be directly under the `:`

File: Formula/faas-cli.rb [hunk 3]
```diff
@@ -19,7 +20,11 @@ def install
     ENV["GOPATH"] = buildpath
     (buildpath/"src/github.com/alexellis/faas-cli").install buildpath.children
     cd "src/github.com/alexellis/faas-cli" do
-      system "go", "build", "-o", bin/"faas-cli"
+      commit = Utils.popen_read("git rev-list -1 HEAD").chomp
+
+      ldflags = "-X main.GitCommit=#{commit}"
+
+      system "go", "build", "-ldflags", ldflags, "-a", "-installsuffix", "cgo", "-o", bin/"faas-cli"
```
Comment (ilovezfs) [inline Formula/faas-cli.rb]:
```
      system "go", "build", "-ldflags", "-X main.GitCommit=#{commit}", "-a",
             "-installsuffix", "cgo", "-o", bin/"faas-cli"
```

Comment (ilovezfs) [inline Formula/faas-cli.rb]: why `-cgo`?

Comment (alexellis) [inline Formula/faas-cli.rb]: It's so it matches the build in the repository.

File: Formula/faas-cli.rb (hunk 2 above)
Comment (alexellis) [inline Formula/faas-cli.rb]: What do you mean?

File: Formula/faas-cli.rb (hunk 3 above)
Comment (ilovezfs) [inline Formula/faas-cli.rb]: ok

File: Formula/faas-cli.rb (hunk 2 above)
Comment (ilovezfs) [inline Formula/faas-cli.rb]:
```
  url "https://github.com/alexellis/faas-cli.git",
    :tag => "0.4.5",
    :revision => "e9c738f9a2833980a1bfc0102b3dfdf0cce2684a"
```
vs.
```
  url "https://github.com/alexellis/faas-cli.git",
      :tag => "0.4.5",
      :revision => "e9c738f9a2833980a1bfc0102b3dfdf0cce2684a"
```

File: Formula/faas-cli.rb (hunk 3 above)
Comment (ilovezfs) [inline Formula/faas-cli.rb]: So just the style change then:
```
      system "go", "build", "-ldflags", "-X main.GitCommit=#{commit}", "-a",
             "-installsuffix", "cgo", "-o", bin/"faas-cli"
```

Comment (alexellis) [inline Formula/faas-cli.rb]: I can't see a difference, what did you change?

File: Formula/faas-cli.rb (hunk 2 above)
Comment (alexellis) [inline Formula/faas-cli.rb]: I can't see a difference unless you mean the whitespace?

File: Formula/faas-cli.rb (hunk 3 above)
Comment (alexellis) [inline Formula/faas-cli.rb]: You mean you want me to wrap the string?

**Comment by alexellis** [comment]
I think we're all good for the Git changeover and your style feedback now. Are you OK to have it merged?

File: Formula/faas-cli.rb (hunk 3 above)
Comment (ilovezfs) [inline Formula/faas-cli.rb]: I inlined the flag and made sure it didn't go passed the eightieth column.

**Review by ilovezfs — APPROVED** [review]

**Review by ilovezfs — CHANGES_REQUESTED** [review]
Is this supposed to be for 0.4.5 or 0.6.0?

**Comment by alexellis** [comment]
It's for 0.4.5 - thanks for picking that up I've updated the title.

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
@alexellis Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**